### PR TITLE
Speed up github workflow by enabling numba jit and test coverage includes only xcube package

### LINKED ISSUE #946: test_gen.py extremely slow on GitHub Actions
**Describe the bug**

GitHub Actions unit test runs are very slow (usually around 40 minutes, versus ~12 minutes per platform on AppVeyor and ~5 minutes on my local machine). Closer inspection of the logs reveals that around 35 of these 40 minutes are spent in one test module -- `test/core/gen/test_gen.py`. For instance, look at the logs for [this](https://github.com/dcs4cop/xcube/actions/runs/8237024354/job/22524820335) test run:

```text
Mon, 11 Mar 2024 17:19:43 GMT test/core/byoa/test_fileset.py ...........                               [  9%]
Mon, 11 Mar 2024 17:19:43 GMT test/core/gen/test_config.py ......                                      [  9%]
Mon, 11 Mar 2024 17:54:01 GMT test/core/gen/test_gen.py ...............                                [ 10%]
Mon, 11 Mar 2024 17:54:01 GMT test/core/gen/test_iproc.py .......                                      [ 11%]
Mon, 11 Mar 2024 17:54:07 GMT test/core/gen2/local/test_generator.py ......                            [ 11%]
Mon, 11 Mar 2024 17:54:07 GMT test/core/gen2/local/test_helpers.py ..                                  [ 11%]
```

**To Reproduce**
1. Run the xcube ‘Unittest and docker builds’ workflow in GitHub Actions, or look at one of the [previous runs](https://github.com/dcs4cop/xcube/actions).
2. Display the logs for the `unittest` step and activate timestamps.
3. Observe that `test_gen.py` takes over half an hour.

**Expected behavior**
`test_gen.py` on GHA completes in seconds or tens of seconds (as it already does on other platforms), rather than in tens of minutes.

### PR DESCRIPTION
**Pull request by konstntokas**
1. Closes #946 unit tests take ~5min which is comparable to local machine (https://github.com/xcube-dev/xcube/actions/runs/9095416416/job/24998584637)
2. Text coverage includes only the xcube package. Note that the text coverage decreases, since the the test folder is not considered anymore in the test coverage report. 
3. github actions are updated. 

Checklist:

* [ ] ~Add unit tests and/or doctests in docstrings~
* [ ] ~Add docstrings and API docs for any new/modified user-facing classes and functions~
* [ ] ~New/modified features documented in `docs/source/*`~
* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [x] AppVeyor CI passes
* [ ] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: .github/workflows/xcube_workflow.yaml [hunk 1]
```diff
@@ -14,11 +14,13 @@ jobs:
   unittest:
     runs-on: ubuntu-latest
     env:
-      NUMBA_DISABLE_JIT: 1
+      NUMBA_DISABLE_JIT: 0
```
Comment (konstntokas) [inline .github/workflows/xcube_workflow.yaml:17]: We could also remove this part completely, depending if we want to have this switch available or not. What do you think?

**Comment by konstntokas** [comment]
Should we remove one of the platforms in appveyor to double the speed?  (https://ci.appveyor.com/project/bcdev/xcube/builds/49819045)

File: .github/workflows/xcube_workflow.yaml (hunk 1 above)
Comment (pont-us) [reply .github/workflows/xcube_workflow.yaml:17]: I think we should keep it -- doesn't do any harm, confirms that numba is not disabled, and saves us looking up the variable name if we need to temporarily disable it for testing in future.

**Review by pont-us — APPROVED** [review]

**Comment by pont-us** [comment]
> Should we remove one of the platforms in appveyor to double the speed, as @pont-us suggested? (https://ci.appveyor.com/project/bcdev/xcube/builds/49819045)

I think this would make sense, since with the speed-up GHA should now be faster than AppVeyor on two platforms. In that case it probably makes the most sense to disable Ubuntu2204 and keep macos, since we're using Ubuntu on GHA. I'd just comment out the Ubuntu2204 item in the "image" section and keep the rest of the configuration in place in case we ever want to re-enable it.

File: .github/workflows/xcube_workflow.yaml (hunk 1 above)
Comment (forman) [reply .github/workflows/xcube_workflow.yaml:17]: Yes, keep it.

**Review by forman — APPROVED** [review]

**Pull request merged**